java/src/main/java/ai/rapids/cudf/ColumnVector.java

```diff
@@ -216,34 +137,38 @@ public ColumnVector(DType type, long rows, Optional<Long> nullCount,
    * @param contiguousBuffer the buffer that this is based off of.
    */
   private ColumnVector(long viewAddress, DeviceMemoryBuffer contiguousBuffer) {
+    super(viewAddress);
     offHeap = new OffHeapState(viewAddress, contiguousBuffer);
     MemoryCleaner.register(this, offHeap);
-    this.type = offHeap.getNativeType();
-    this.rows = offHeap.getNativeRowCount();
     // TODO we may want to ask for the null count anyways...
     this.nullCount = Optional.empty();
 
     this.refCount = 0;
     incRefCountInternal(true);
   }
 
-  static ColumnVector fromViewWithContiguousAllocation(long columnViewAddress, DeviceMemoryBuffer buffer) {
-    return new ColumnVector(columnViewAddress, buffer);
+  /** Creates a ColumnVector from a column view handle
+   * @return a new ColumnVector
+   */
+  public static ColumnVector fromColumnView(long columnViewHandle) {
+    long mainCvHandle = ColumnView.copyColumnViewToCV(columnViewHandle);
+    return new ColumnVector(mainCvHandle);
   }
 
-  /**
-   * Returns a column of strings where, for each string row in the input,
-   * the first character after spaces is modified to upper-case,
-   * while all the remaining characters in a word are modified to lower-case.
-   *
-   * Any null string entries return corresponding null output column entries
-   */
-  public ColumnVector toTitle() {
-    assert type == DType.STRING;
-    return new ColumnVector(title(getNativeView()));
+
+  private static long initViewHandle(DType type, int rows, int nc, DeviceMemoryBuffer dataBuffer,
+                                     DeviceMemoryBuffer validityBuffer, DeviceMemoryBuffer offsetBuffer, long[] childHandles) {
+    long cd = dataBuffer == null ? 0 : dataBuffer.address;
+    long cdSize = dataBuffer == null ? 0 : dataBuffer.length;
+    long od = offsetBuffer == null ? 0 : offsetBuffer.address;
+    long vd = validityBuffer == null ? 0 : validityBuffer.address;
+    return makeCudfColumnView(type.typeId.getNativeId(), type.getScale(), cd, cdSize,
+        od, vd, nc, rows, childHandles) ;
   }
 
-  private native long title(long handle);
+  static ColumnVector fromViewWithContiguousAllocation(long columnViewAddress, DeviceMemoryBuffer buffer) {
```

Comment:
why not just make the constructor package private?  Unless we want to move towards factory methods as the main entry point for consistency.

Reply:
I kept this as is since it was not really related to nested types, sorry if I should. I can get it in this patch if necessary

Reply:
It is just a little ugly, and I think we need some follow on work to decide if we are going to stick with public constructors or if we want to move towards factory methods.  I am leaning more and more towards factory methods.